src/main/scala/cognite/spark/DefaultSource.scala

```diff
@@ -95,18 +95,25 @@ class DefaultSource
         .getOrElse(throw new IllegalArgumentException(
           s"$onConflictName is not a valid onConflict option. Please choose one of the following options instead: ${OnConflict.values
             .mkString(", ")}"))
+    val parallelismPerPartition = {
+      val inputOrDefault = toPositiveInt(parameters, "parallelismPerPartition").getOrElse(
+        Constants.DefaultParallelismPerPartition)
+      if (inputOrDefault > partitions) { inputOrDefault } else { partitions }
```

Comment:
Hmm, why?

Reply:
Now that all filters for a partition are merged into one stream we'll get one or more empty `Iterator`s when grouping the streams in `.compute` if `parallelismPerPartition > partitions`. This avoids that. I suppose `max(parallelismPerPartitions, partitions) is nicer though, and I'll add a comment.

Reply:
I realize it's also pretty confusing that `partitions` is related to the partitions of a CDF endpoint, while `parallelismPerPartition` is related to Spark partitions. We'll need to use another name for both.